### PR TITLE
feat: save sql charts

### PR DESCRIPTION
**Pull request by rephus**
Closes:   #10639

### Description:

Adds SQL chart endpoints:
- get
- create
- update 
- delete

Screenshot 2024-07-11 at 15 18 20

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/models/SavedSqlModel.ts [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+        },
+    ): Promise<string> {
+        const [{ saved_sql_version_uuid: savedSqlVersionUuid }] = await trx(
+            SavedSqlVersionsTableName,
+        ).insert(
+            {
+                saved_sql_uuid: data.savedSqlUuid,
+                sql: data.sql,
+                config: data.config,
+                chart_kind: ChartKind.VERTICAL_BAR, // todo: get chart kind from config
```
Comment (magnew) [inline packages/backend/src/models/SavedSqlModel.ts:229]: I spent longer than I wanna admit thinking I was getting something wrong in the FE cause everything kept getting rendered as bar charts 😂 

That'll still be the case until we update these

File: packages/backend/src/services/SavedSqlService/SavedSqlService.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        }
+        const space = await this.spaceModel.getSpaceSummary(sqlChart.spaceUuid);
+        const { isPrivate } = space;
+        const access = await this.spaceModel.getUserSpaceAccess(
+            user.userUuid,
+            sqlChart.spaceUuid,
+        );
+        if (
+            user.ability.cannot(
+                'create',
```
Comment (magnew) [inline packages/backend/src/services/SavedSqlService/SavedSqlService.ts:95]: The FE lets them send these requests for now. This access check means its 'safe' but we can disable saving them if they done have this.

File: packages/common/src/types/sqlRunner.ts [hunk 3]
```diff
@@ -14,3 +20,71 @@ export type SqlRunnerBody = {
...
+
+export type ApiSqlChart = {
+    status: 'ok';
+    results: SqlChart;
+};
+
+export type ApiCreateSqlChart = {
+    status: 'ok';
+    results: {
+        savedSqlUuid: string;
```
Comment (magnew) [inline packages/common/src/types/sqlRunner.ts:80]: This is all we need, does it need a type? I just stubbed it out like this in the API call

**Review by magnew — APPROVED** [review]
🔥 Looks great. The best review I can give this is I wrote a bunch of code against it today. The only real issue I hit was the chart type default.

File: packages/backend/src/models/SavedSqlModel.ts (hunk 1 above)
Comment (ZeRego) [reply packages/backend/src/models/SavedSqlModel.ts:229]: Fixed in the UI PR as we have the final config types there 👍

File: packages/common/src/types/sqlRunner.ts (hunk 3 above)
Comment (ZeRego) [reply packages/common/src/types/sqlRunner.ts:80]: Would it return the same chart kind type as the one sent in the body? My idea was for the FE to change the URL and fetch the saved SQL with the GET endpoint.

**Pull request merged**